CHANGELOG.md

```diff
@@ -4,6 +4,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 ## Unreleased
 ### Added
 ### Changed
+- [BC Break] No longer extending `OpenSearch::Client` from `opensearch-ruby` in favor of simply providing a SigV4 request signer as described in [this guide](https://github.com/opensearch-project/opensearch-ruby/blob/main/DEVELOPER_GUIDE.md#create-a-request-signer).
```

Comment:
I would say "Compatibility with opensearch-ruby 4.0", which is really the change being made here. Everything else is implementation detail.

Reply:
Sounds good. But I think we should explain this version breaks BC, meaning users may need to change their existing code after installing it. Where would it be mentioned?

Reply:
You should put this into an UPGRADING.md file with a clear before/after. See https://github.com/opensearch-project/opensearch-ruby/blob/main/UPGRADING.md for an example.

Reply:
Thank you for the guidance! I've added UPGRADING.md.

Reply:
@dblock do you mean
```ruby
signer = Aws::Sigv4::Signer.new(...)

client = OpenSearch::Client.new(
   ...
   request_signer: signer
)
```
This gem can now be decoupled from the main gem as it only needs to provide a signing object with the `sign_request` method. We no longer have to worry about the inner working of `opensearch-ruby` gem and just follow the interface described in the [dev guide](https://github.com/opensearch-project/opensearch-ruby/blob/main/DEVELOPER_GUIDE.md#create-a-request-signer) of the main gem for a request signer.